app/models/archive_endpoint.rb

```diff
@@ -16,6 +16,23 @@ class ArchiveEndpoint < ApplicationRecord
   # TODO: after switching to string, validate that input resolves to class which #is_a class of the right type?
   validates :delivery_class, presence: true
 
+  # for the given druid, which archive endpoints should have archive copies?
+  scope :archive_targets, lambda { |druid|
+    joins(preservation_policies: [:preserved_objects]).where(preserved_objects: { druid: druid })
+  }
+
+  scope :which_have_archive_copy, lambda { |druid, version|
+    joins(archive_preserved_copies: [:preserved_object])
+      .where(
+        preserved_objects: { druid: druid },
+        archive_preserved_copies: { version: version }
+      )
+  }
+
+  scope :which_need_archive_copy, lambda { |druid, version|
```

Comment:
maybe add comment:

"for a druid, version, which zip_endpoints need a zipped_moab_version, per preservation_policy"